src/components/App.tsx

```diff
@@ -2386,8 +2386,15 @@ class App extends React.Component<AppProps, AppState> {
 
   private redirectToLink = () => {
     if (
-      this.lastPointerDown!.clientX !== this.lastPointerUp!.clientX ||
-      this.lastPointerDown!.clientY !== this.lastPointerUp!.clientY
+      (!this.isMobile &&
```

Comment:
can you reverse the conditions, first ` this.isMobile && distance() || (strict check), for better readability
Or it could be further simpliefied to just checking the distance =>
store distance in  a variable and check if its greater than threshold when mobile or 0 otherwise for strict check

Reply:
I will, but probably only tonight. Right now I am not at my desktop, and linting using just the GitHub editor is a pain.
Also, I need to understand what isMobile covers. Are iPads included?
Finally, any idea why the link is not working on an iPad (I am testing on iOS 15.1 and links don't work, even with this change)?

Reply:
ipads are not included in mobile https://github.com/excalidraw/excalidraw/blob/master/src/components/App.tsx#L891, I pushed a fix, could you confirm if it works in ipad ?

Reply:
With this latest change it stopped working on my Android as well. To confirm, right now the link click does not work on my Android 9 mobile and my iOS 15.1 iPad.

Reply:
I pushed a change to correct the logic. Now its working on my Android phone, but still not working on the iPad.

Reply:
was your initial change (with out conditions for threshold) working for ipad ?

Reply:
no, it was not. I just tried to push another change… but still no success on the iPad

Reply:
Now it works... but I simply removed check for isMobile || isIPad.
It turns out `export const isIPad = /iPad/.test(window.navigator.platform);` resolves false on iOS15.1 (in my case).
I see no negative effect of removing the check for mobile and iPad and allowing link click on desktop as well. Why was it limited to isMobile?

Reply:
@dwelle, btw, my finding above also suggests that probably isDarwin resolves to false on iOS15.1 as well. Back to the conversation from yesterday, these device checks are flimsy. Maybe would be better to use a package like [platform.js](https://www.npmjs.com/package/platform) - thought this one seems to be dead, it was last updated two years ago.

Reply:
> I see no negative effect of removing the check for mobile and iPad and allowing link click on desktop as well. Why was it limited to isMobile?

Yes, link-click can have the threshold on desktop as well, but on the link icon only (not the element body — which on Mobile is disabled, anyway)

> Maybe would be better to use a package like [platform.js](https://www.npmjs.com/package/platform) - thought this one seems to be dead, it was last updated two years ago.

I'm not sure, and as you said, it's not up-to-date. It purportedly [detects iPad as Mac](https://github.com/bestiejs/platform.js/issues/194).

Reply:
The `isMobile` check in `isPointHittingLinkIcon` was added to make sure that clicking on element takes to link only in view mode in non mobile devices, this means that for `ipad` also in view mode the element will be clickable.